### DISCUSSION TITLE
Style selector for the root of a ListBoxItem

### DISCUSSION THREAD

**Discussion by oli-g-sk** [AvaloniaUI/Avalonia#15326] [Q&A] [answered]
I'm trying to create a style that targets the root of any `ListBoxItem` with an arbitrary `ItemTemplate`.

Basically, I want any root control rendered in any `ListBoxItem` to always have a transparent (and not null) `Background` by default, so that pointer events always work.

```xml
<Style Selector="ListBoxItem > ContentPresenter > :is(Panel)">
    <Setter Property="Background" Value="Transparent" />
</Style>
<Style Selector="ListBoxItem > ContentPresenter > :is(Border)">
    <Setter Property="Background" Value="Transparent" />
</Style>
```

Unless I'm misunderstanding the syntax, this should select the direct `ContentPresenter` child of a `ListBoxItem` and follow to the direct child if it's of a specified base type (something that has a `Background` property).

Am I doing something wrong, or is the combo of the "direct child" & "derived classes" features not supported? Thank you!

**Comment by maxkatz6** [comment, marked as answer]
">" selector targets **logical** child, but ContentPresenter is part of the visual sub tree, specifically in the ListBoxItem control template.
So, you need a `/template/` selector there instead. 

And since you want to target panel or border inside of the ContentPresenter, you probably can even write something like:
```
<Style Selector="ListBoxItem > :is(Border)">
    <Setter Property="Background" Value="Transparent" />
</Style>
```

Since Border would be a **logical child** of the ListBoxItem.

**Reply by timunie** [reply]
Use DevTools F12 to inspect visual tree

**Reply by oli-g-sk** [reply]
@timunie Yeah I know about that - but since you mentioned it, just gotta say that F12 tool is pretty neat! 👏

@maxkatz6 Thank you! For some reason it didn't even occur to me that it targets _logical_ children, now it makes so much sense!

**Reply by oli-g-sk** [reply]
@maxkatz6 One more question - your suggestion _did_ help ("skipping" the `ContentPresenter` since it's visual only), but...
If `>` targets the direct _logical_ child, how come the following style _does_ work?

```xml
<Style Selector="ListBoxItem">
    <Style Selector="^:selected > ContentPresenter">
        <Setter Property="Background" Value="Red" />
    </Style>
</Style>
```
